addons/website/controllers/main.py

```diff
@@ -303,6 +305,26 @@ def website_configurator(self, step=1, **kwargs):
             return request.redirect('/')
         if request.env.lang != request.website.default_lang_id.code:
             return request.redirect('/%s%s' % (request.website.default_lang_id.url_code, request.httprequest.path))
+
+        # Check that user is not refreshing the configuration webpage after starting website configuration
+        cr = request.env.cr
+        try:
+            with mute_logger('odoo.sql_db'):
+                cr.execute("SELECT * FROM ir_module_module FOR UPDATE NOWAIT")
```

Comment:
I am not sure about all details, but intuitively Locking a table in this way is strange and unnecessary. 
It will add lock for each row and return each row
Even if we have to use Lock, why not directly lock the table?

Reply:
The purpose of this code is not to lock, but to check if is there any locked row.

Reply:
I think they are the same `FOR UPDATE` will add row lock until the end of this transaction. I think you can lock the table in the  `Exclusive` mode with `NOWAIT`. The only difference is in your code, you allow another transaction to `Insert` ir.module.module rows, which I think is useless.